core/src/apps/zcash/get_viewing_key.py

```diff
@@ -0,0 +1,46 @@
+from typing import TYPE_CHECKING
+
+from trezor import ui
+from trezor.enums import ButtonRequestType
+from trezor.messages import ZcashGetViewingKey, ZcashViewingKey
+from trezor.ui.layouts import confirm_action
+
+from apps.common import coininfo
+
+from .orchard.keychain import with_keychain
+from .unified import Typecode, encode_fvk, encode_ivk
+
+if TYPE_CHECKING:
+    from trezor.wire import Context
+    from .orchard.keychain import OrchardKeychain
+
+
+@with_keychain
+async def get_viewing_key(
+    ctx: Context, msg: ZcashGetViewingKey, keychain: OrchardKeychain
+) -> ZcashViewingKey:
+    await require_confirm_export_viewing_key(ctx, msg)
```

Comment:
please add `await paths.validate_path(ctx, keychain, msg.z_address_n)` before this line

Reply:
7a67e9a8ad74eba5cfae7e042fe983683f5c8946